### ISSUE TITLE
LoadTaskFromJs plugin doesn't correctly import js modules compiled from TypeScript

### ISSUE THREAD

**Issue by confusedmatrix** [graphile/worker#416]
When trying to load a task (written in TypeScript) and compiled to js, I get the following error:

```
Error: Error processing 'root/tasks/hello.js': name is not defined
    at loadTaskFromFiles (/root/node_modules/graphile-worker/dist/plugins/LoadTaskFromJsPlugin.js:53:27)
    at async getTasksInternal (/root/node_modules/graphile-worker/dist/getTasks.js:85:13)
    at async main (/root/node_modules/graphile-worker/dist/cli.js:113:30)
```

I believe the root cause however is [this line](https://github.com/graphile/worker/blob/188c330e76e036bf2671e772c9b86b93beb5b31b/src/plugins/LoadTaskFromJsPlugin.ts#L45C1-L45C48), checking for a single exported property when determining whether to load `rawMod.default`.

My simple task in typescript is:

```
export default async (payload, helpers) => {
    helpers.logger.debug(`Received ${JSON.stringify(payload)}`);
};
```

compiled to js using `tsc tasks/*.ts --module=nodenext` results in:

```
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = async (payload, helpers) => {
    helpers.logger.debug(`Received ${JSON.stringify(payload)}`);
};
```

As you can see typescript adds the `__esModule` module property on the `exports` object, which means 2 properties are exported (`__esModule` and `default`).

**Comment by benjie** [comment]
```
> rawMod = await import('./test.js')
[Module: null prototype] {
  __esModule: true,
  default: { default: [AsyncFunction (anonymous)] }
}
```
Looks like there's two layers of defaults in this. And in fact `rawMod.__esModule` is true, but that's not the one in your source; that's actually `rawMod.default.__esModule`

Looks like we need _another_ bit of special ESM handling  :roll_eyes:

**Comment by benjie** [comment]
Feel free to raise a PR with a reproducing test (and, ideally, a fix)

**Comment by alexghattas** [comment]
Hi all, chiming in here to just say I am experiencing the same issue.

Does this mean, that without a fix, we cannot use TS in the task executors?

**Comment by benjie** [comment]
Fix is out in v0.16.1
